model_zoo/tf_estimator/estimator_executor/conf.py

```diff
@@ -11,9 +11,90 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+#  define your own custome reader
+import numpy as np
 from MyEstimator import MyEstimator
 
+from dlrover.python.elastic_agent.sharding.client import ShardingClient
 from dlrover.trainer.tensorflow.util.column_info import Column
+from dlrover.trainer.tensorflow.util.reader_util import reader_registery
+
+
+def build_data_shard_service(
+    batch_size=1,
+    num_epochs=1,
+    dataset_size=1,
+    num_minibatches_per_shard=1,
+    dataset_name="iris_training_data",
+):
+    sharding_client = ShardingClient(
+        dataset_name=dataset_name,
+        batch_size=batch_size,
+        num_epochs=num_epochs,
+        dataset_size=dataset_size,
+        num_minibatches_per_shard=num_minibatches_per_shard,
+    )
+    return sharding_client
+
+
+class FakeReader:
```

Comment:
It is better to define an abstract class as the interface. Users can implement their readers by inheriting the abstract class. For example:

```Python
from abc import ABCMeta, abstractmethod

class ElasticReader(metaclass=ABCMeta):
    def __init__(self, path, num_epochs, batch_size):
         ...
    @abstractmethod
    def count(self):
       pass

    @abstractmethod
    def read(self, start_index, end_index):
          pass

    def iterator(self):
        while True:
            shard = self._data_shard_client.fetch_shard()
            if shard is None:
                break
            for data in self.read(shard.start, shard.end):
                yield data



```